object.go

```diff
@@ -965,7 +965,16 @@ func (o *objResult) Read(p []byte) (n int, err error) {
 	}
 
 	r := o.r.(net.Conn)
-	r.SetReadDeadline(time.Now().Add(2 * time.Second))
+	deadline, ok := o.ctx.Deadline()
+	if !ok {
+		deadline = time.Now().Add(2 * time.Second)
+	}
+
+	err = r.SetReadDeadline(deadline)
```

Comment:
This is an internal pipe IIRC.. So I wonder if you are getting a slow consumer internally in your app from the NATS server you are connected to, or the NATS.go client is deeming this a slow consumer?